src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java

```diff
@@ -154,7 +154,7 @@ protected SingleChronicleQueue(@NotNull final SingleChronicleQueueBuilder builde
             fileAbsolutePath = path.getAbsolutePath();
             wireType = builder.wireType();
             blockSize = builder.blockSize();
-            overlapSize = Math.max(64 << 10, builder.blockSize() / 4);
+            overlapSize = Math.min(Math.max(64 << 10, builder.blockSize() / 4), 1L << 30);
```

Comment:
`1L << 30` could be a constant or returned by a method.

Reply:
+1 for constant